src/components/Grid/Grid.tsx

```diff
@@ -0,0 +1,140 @@
+import React, {
+  Children as ReactChildren,
+  createElement,
+  cloneElement,
+  CSSProperties,
+  Fragment,
+  useEffect,
+  useMemo,
+  useState,
+  createRef,
+  PropsWithChildren,
+} from "react";
+import { forwardRef } from "react";
+import { Children } from "~/utils/typing/children";
+import { SPACE_UNIT } from "~/utils/constants";
+import { Box } from "~/index";
+
+export type ContainerProps = PropsWithChildren<{
+  "data-test-id"?: string;
+  rows: number;
+  cols: number;
+  rowsGap?: number;
+  colsGap?: number;
+  className?: string;
+  style?: CSSProperties;
+}> & {
+  children: React.ReactElement | React.ReactElement[];
+};
+
+const Container = forwardRef<HTMLDivElement, ContainerProps>((props, ref) => {
+  const { children } = props;
+  const [invalidError, setInvalidError] = useState(false);
+  const errorMessage =
+    "invalid child type, all children should be <Grid.Item />";
+
+  const cloned = useMemo(
+    () =>
+      React.Children.map(children, (child) =>
+        cloneElement(child, { ref: createRef() })
+      ),
+    [children]
+  );
+
+  useEffect(() => {
+    if (Array.isArray(cloned)) {
+      let count = 0;
+      cloned.forEach((child) => {
```

Comment:
I would safely assume you could use [Array.some()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/some) here

Reply:
done